allensdk/core/sitk_utilities.py

```diff
@@ -0,0 +1,172 @@
+
+    Returns
+    -------
+    dict : 
+        Extracted information.
+
+    '''
+
+    return {'spacing': image.GetSpacing(),
+            'origin': image.GetOrigin(),
+            'size': image.GetSize(),
+            'direction': image.GetDirection(),
+            'ncomponents': image.GetNumberOfComponentsPerPixel()}
+
+
+def set_sitk_image_information(image, information):
+    ''' Set information on a SimpleITK image
+
+    Parameters
+    ----------
+    image : sitk.Image
+        Set information on this image.
+    information : dict
+        Stores information to be set
+
+    '''
+
+    if 'spacing' in information:
+        image.SetSpacing(information.pop('spacing'))
+    if 'origin' in information:
+        image.SetOrigin(information.pop('origin'))
+    if 'direction' in information:
+        image.SetDirection(information.pop('direction'))
+
+    if 'size' in information:
+        assert(np.array_equal( information.pop('size'), image.GetSize() ))
+    if 'ncomponents' in information:
+        assert( information.pop('ncomponents') == image.GetNumberOfComponentsPerPixel() )
+    
+    if not len(information) == 0:
+        warnings.warn('unwritten keys: {}'.format(','.join(information.keys())))
+
+
+def fix_array_dimensions(array, ncomponents=1):
+    ''' Convenience function that reorders ndarray dimensions for io with SimpleITK
+
+    Parameters
+    ----------
+    array : np.ndarray
+        The array to be reordered
+    ncomponents : int, optional
+        Number of components per pixel, default 1. 
+
+    Returns
+    -------
+    np.ndarray : 
+        Reordered array
+
+    '''
+
+    act_size = list(array.shape)
+    ndims = len(act_size)
+    multicomponent = ncomponents > 1
+
+    from_order = list(range( ndims - multicomponent ))
+    to_order = list(range( ndims - multicomponent ))[::-1]
+
+    if multicomponent:
+        from_order += [-1]
+        to_order += [-1]
+
+    return np.ascontiguousarray(np.moveaxis(array, from_order, to_order))
+
+
+def read_ndarray_with_sitk(path):
+    ''' Read a numpy array from a file using SimpleITK
+
+    Parameters
+    ----------
+    path : str
+        Read from this path
+    
+    Returns
+    -------
+    image : np.ndarray
+        Obtained array
+    information : dict
+        Additional information about the array
+
+    '''
+
+    image = sitk.ReadImage(str(path))
+    information = get_sitk_image_information(image)
+    image = sitk.GetArrayFromImage(image)
+
+    image = fix_array_dimensions(image, information['ncomponents'])
+    return image, information
+
+
+def write_ndarray_with_sitk(array, path, **information):
+    ''' Write a numpy array to a file using SimpleITK
+
+    Parameters
+    ----------
+    array : np.ndarray
+        Array to be written.
+    path : str
+        Write to here
+    **information : dict
+        Containes additional information to be stored in the image file. 
+
+    '''
+
+    if not 'ncomponents' in information:
```

Comment:
Maybe a mention in the docstring of what ncomponents is, or in an example somewhere.